api/enki.py

```diff
@@ -539,226 +479,253 @@ def queue_response(self, status_code, headers={}, content=None):
-    not normally necessary because the Enki API combines
-    bibliographic and availability data.
+class BibliographicParser(object):
+    """Parses Enki's representation of book information into
+    Metadata and CirculationData objects.
     """
 
-    SERVICE_NAME = "Enki Bibliographic Coverage Provider"
-    DATA_SOURCE_NAME = DataSource.ENKI
-    PROTOCOL = EnkiAPI.ENKI
-    INPUT_IDENTIFIER_TYPES = EnkiAPI.ENKI_ID
-
-    def __init__(self, collection, api_class=EnkiAPI, **kwargs):
-        """Constructor.
-
-        :param collection: Provide bibliographic coverage to all
-            Enki books in the given Collection.
-        :param api_class: Instantiate this class with the given Collection,
-            rather than instantiating EnkiAPI.
-        """
-        _db = Session.object_session(collection)
-        super(EnkiBibliographicCoverageProvider, self).__init__(
-            collection, **kwargs
-        )
-        if isinstance(api_class, EnkiAPI):
-            # We were given a specific EnkiAPI instance to use.
-            self.api = api_class
-        else:
-            self.api = api_class(_db, collection)
-        self.parser = BibliographicParser()
-
-    def process_batch(self, identifiers):
-        identifier_strings = self.api.create_identifier_strings(identifiers)
-        response = self.api.availability(title_ids=identifier_strings)
-        seen_identifiers = set()
-        batch_results = []
-        for metadata, availability in self.parser.process_all(response.content):
-            identifier, is_new = metadata.primary_identifier.load(self._db)
-            if not identifier in identifiers:
-                # Enki told us about a book we didn't ask
-                # for. This shouldn't happen, but if it does we should
-                # do nothing further.
-                continue
-            seen_identifiers.add(identifier.identifier)
-            result = self.set_metadata(identifier, metadata)
-            if not isinstance(result, CoverageFailure):
-                result = self.handle_success(identifier)
-            batch_results.append(result)
-
-        # Create a CoverageFailure object for each original identifier
-        # not mentioned in the results.
-        for identifier_string in identifier_strings:
-            if identifier_string not in seen_identifiers:
-                identifier, ignore = Identifier.for_foreign_id(
-                    self._db, api.ENKI_ID, identifier_string
-                )
-                result = CoverageFailure(
-                    identifier, "Book not found in Enki", data_source=self.output_source, transient=True
-                )
-                batch_results.append(result)
-        return batch_results
-
-    def process_item(self, identifier):
-        results = self.process_batch([identifier])
-        return results[0]
-
-    def handle_success(self, identifier):
-        return self.set_presentation_ready(identifier)
```

Comment:
The only real downside of getting rid of the BibliographicCoverageProvider is that the process of creating a work and making it presentation-ready had to be moved into the Monitor.